src/openforms/registrations/contrib/email/config.py

```diff
@@ -91,3 +94,30 @@ class Meta:
                 "email_content_template_text",
             ),
         ]
+
+
+class Options(TypedDict):
+    """
+    Shape of the email registration plugin options.
+
+    This describes the shape of :attr:`EmailOptionsSerializer.validated_data`, after
+    the input data has been cleaned/validated.
+    """
+
+    to_emails: list[str]
+    attachment_formats: NotRequired[list[AttachmentFormat | str]]
```

Comment:
Why not just `list[AttachmentFormat]`?

Reply:
because the key is absent in `serializer.validated_data` if you do this:

```py
>>> from openforms.registrations.contrib.email.config import *
>>> serializer = EmailOptionsSerializer(data={"to_emails": []})
>>> serializer.is_valid()
True
>>> serializer.validated_data
{'to_emails': [], 'attach_files_to_email': None}

```

Reply:
Oh, misunderstood! it's about the `AttachmentFormat` vs. `str` union - restoring from the JSON in the DB will have strings and not the enum at runtime. They pass equality checks, but they are a different type.

Ideally we can produce a `Literal[...]` from an enum, but that's currently not possible in Python.